http/client/client.go

```diff
@@ -19,23 +19,59 @@
 package client
 
 import (
+	"bytes"
 	"crypto/tls"
 	"errors"
+	"fmt"
+	"github.com/nuts-foundation/nuts-node/core"
+	"io"
 	"net/http"
 	"time"
 )
 
+// SafeHttpTransport is a http.Transport that can be used as a default transport for HTTP clients.
+var SafeHttpTransport *http.Transport
+
 func init() {
-	httpTransport := http.DefaultTransport.(*http.Transport)
-	if httpTransport.TLSClientConfig == nil {
-		httpTransport.TLSClientConfig = &tls.Config{}
+	SafeHttpTransport = http.DefaultTransport.(*http.Transport).Clone()
+	if SafeHttpTransport.TLSClientConfig == nil {
+		SafeHttpTransport.TLSClientConfig = &tls.Config{}
 	}
-	httpTransport.TLSClientConfig.MinVersion = tls.VersionTLS12
+	SafeHttpTransport.TLSClientConfig.MinVersion = tls.VersionTLS12
+	// to prevent slow responses from public clients to have significant impact (default was unlimited)
+	SafeHttpTransport.MaxConnsPerHost = 5
+
+	DefaultCachingTransport = SafeHttpTransport
```

Comment:
`Engine.configureClient` also sets `DefaultCachingTransport` and comes after this. Does this do anything? Maybe add a comment that the real value is set somewhere else